docs/readmes/proposals/p024_magma_settlement_service.md

```diff
@@ -0,0 +1,281 @@
+- Step [7] The Helium Handler updates State Channel with signed data usage received on Step [6]
+- Step [8] (Optional) The Helium Handler sends FreedomFi balance update to Session Purchaser.
+- Step [9] Delete session response from SGW to Home PGW to indicate the PDN connection is closed.
+- Step [10-11] MME/eNodeB closes radio link for the UE.
+
+#### 3.2.6 TAP-Out procedure
+
+A roaming agreement between the home network operator and the visited network operator defines the terms that enable each other's customers access to the wireless networks. The visited network operator records the activities performed by the roaming subscriber and then sends the call event details to the home network operator in the format agreed upon in the roaming agreement, usually Transferred Account Procedure (TAP) format. TAP is the process that allows a visited network operator to send call event detail records of roaming subscribers to their respective home network operators to be able to bill for the subscriber's roaming usage.
+
+#### 3.2.8 RAP-In procedure
+
+The home network operator validates the data in the TAP files to ensure that it conforms to the TAP standard and to the terms of the roaming agreement. If the received TAP file contains any errors, the home network operator can reject the entire file or only the incorrect call event detail records. The incorrect file or records are returned to the visited network operator in a Returned Account Procedure (RAP) file.
+RAP process is used to return rejected TAP files and records to the visited network operator for corrections. A RAP file contains the rejected TAP file or records and additional data about the error, such as the error code or the error cause. The visited network operator corrects the errors and sends the corrected TAP file back to the home network operator.
+
+### 3.3 Integration interfaces
+
+The following figure shows the integration interfaces between Magma domain and OCS /DCH.
+
+![Integration interfaces](https://user-images.githubusercontent.com/93994458/149514277-062e0834-cb8c-49af-b0e4-867c4664c694.png)
+
+#### 3.3.1 DIAMETER Gy interface specification
+
+Interface transport protocol: TCP/Diameter/DCCA.
+The 3GPP based Gy interface between Session Proxy and Session Purchaser to implement real-time quota management for the packet data service for the roaming subscribers. While this interface resembles Gy in almost every way, this is a non-standard use of the Gy protocol defined to allow accounting at the operator level and allow the session purchaser a control point to approve or deny sessions at initiation or at update points throughout a session.
+The Session Purchaser is the Diameter Credit Control server, which provides the online charging data to the Session Proxy based on volume quota.
+The connection between the Session Proxy (client) and Session Purchaser (server) is TCP based. There are a series of message exchanges to manage quota in real-time.
```

Comment:
My suggestion is Session Purchaser to be implemented as a proper OCS, meaning, it should return CCA (it can be a hardcoded default resonse).  But that may allow you to use the system as is (sessiond and session proxy)

[here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Fservices%2Ftestcore%2Focs%2Fmock_ocs%2Fccr_handler.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=28&start_col=4&end_row=28&end_col=4) you have a mock OCS. that may easily be extended to be your session purchaser

Reply:
absolutely we'll use standard OCS as session purchaser